gunagala/imager.py

```diff
@@ -1561,18 +1579,19 @@ def make_noiseless_image(self,
         """
         electrons = np.zeros((self.wcs._naxis2,
                               self.wcs._naxis1)) * u.electron / (u.second * u.pixel)
+        self.set_WCS_centre(centre, **kwargs)
 
         # Calculate observed sky background
         sky_rate = self.sky_rate[filter_name]
         if hasattr(self.sky, 'relative_brightness'):
-            pixel_coords = self.get_pixel_coords(centre)
+            pixel_coords = self.get_pixel_coords()
             relative_sky = self.sky.relative_brightness(pixel_coords, obs_time)
             sky_rate = sky_rate * relative_sky
         electrons = electrons + sky_rate
 
         if stars is not None:
             for (coords, magnitude) in stars:
-                coords = SkyCoord(coords)
+                coords = SkyCoord(coords, **kwargs)
```

Comment:
Passing the same set of keyword args to `set_WCS_centre` and the `SkyCoord` constructor for the star coordinates is forcing the user to use the same (or at least similar) formats for both sets of coordinates.

You could avoid this limitation by giving `make_noiseless_image` two optional `dict` arguments, e.g. `centre_kwargs` and `star_kwargs`, which default to empty dictionaries. Then can pass them on with `self.set_WCS_centre(centre, **centre_kwargs)` and `coords = SkyCoord(coords, **star_kwargs)`

Reply:
Interesting idea. I tried implementing @AnthonyHorton . Let me know if this is what you had in mind.